canvas/views/views.py

```diff
@@ -78,3 +81,25 @@ def events_options_view(request):
     return render(request, 'canvas/course_event_options.html', {
         'events': course.events.all(),
     })
+
+
+def create_event_view(request):
+    if request.method == 'POST':
+        form = CreateEventForm(request.POST)
+        if form.is_valid():
+            try:
+                create_event(name=form.cleaned_data['name'], course=form.cleaned_data['course'],
+                             count_for_tokens=form.cleaned_data['count_for_tokens'],
+                             start_date=form.cleaned_data['start_datetime'],
+                             end_date=form.cleaned_data['end_datetime'])
+                messages.add_message(request, messages.SUCCESS, 'Event created successfully')
+            except EventCreateException:
+                messages.add_message(request, messages.ERROR, 'Error in event creation.')
+
+            return HttpResponseRedirect('.')
```

Comment:
I'm not sure what '.' does in HttpResponseRedirect. I don't think you need to redirect here you have to stay on the same page to display errors in the form.